packages/react/src/components/SplitButton/SplitButton.tsx

```diff
@@ -0,0 +1,166 @@
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+import { Ref } from '@stardust-ui/react-component-ref'
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+import * as _ from 'lodash'
+
+import {
+  WithAsProp,
+  withSafeTypeForAs,
+  ComponentEventHandler,
+  ShorthandValue,
+  ShorthandCollection,
+} from '../../types'
+import {
+  UIComponent,
+  UIComponentProps,
+  ChildrenComponentProps,
+  ContentComponentProps,
+  applyAccessibilityKeyHandlers,
+  commonPropTypes,
+} from '../../lib'
+import { Accessibility } from '../../lib/accessibility/types'
+import Button, { ButtonProps } from '../Button/Button'
+import MenuButton from '../MenuButton/MenuButton'
+import { splitButtonBehavior } from '../../lib/accessibility'
+import { MenuProps } from '../Menu/Menu'
+import { MenuItemProps } from '../Menu/MenuItem'
+
+export interface SplitButtonProps
+  extends UIComponentProps,
+    ChildrenComponentProps,
+    ContentComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   */
+  accessibility?: Accessibility
+
+  /** Element to be rendered in-place where the popup is defined. */
+  button?: ShorthandValue<ButtonProps>
+
+  /** Element to be rendered in-place where the popup is defined. */
+  menu?: ShorthandValue<MenuProps> | ShorthandCollection<MenuItemProps>
+
+  /**
+   * Called after user's click.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onClick?: ComponentEventHandler<ButtonProps>
+
+  /** A split button can be formatted to show different levels of emphasis. */
+  primary?: boolean
+
+  /** A split button can be formatted to show different levels of emphasis. */
+  secondary?: boolean
+}
+
+class SplitButton extends UIComponent<WithAsProp<SplitButtonProps>> {
+  static create: Function
+
+  static displayName = 'SplitButton'
+
+  static className = 'ui-splitbutton'
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    button: customPropTypes.itemShorthand,
+    menu: PropTypes.oneOfType([customPropTypes.itemShorthand, customPropTypes.itemShorthand]),
+    onClick: PropTypes.func,
+    onToggle: PropTypes.func,
```

Comment:
don't `onToggle` defined in props and used

Reply:
addressed